src/pkg/packager/publish.go

```diff
@@ -38,6 +40,11 @@ func PublishFromOCI(ctx context.Context, src registry.Reference, dst registry.Re
 	l := logger.From(ctx)
 	start := time.Now()
 
+	// Allow zero but other sanitize retries if less-than zero
+	if opts.Retries < 0 {
+		opts.Retries = config.ZarfDefaultRetries
+	}
```

Comment:
retry-go does allow 0 in retry.Attempts(), but it will retry infinitely until the function succeeds. Given that 0 will be the default at the SDK layer I believe we should change this. 

We might also want this to default to 1 instead of 3 as it's consistent with the CLI and the most common user flow.